### PR TITLE
Dimensions pull

### PR DESCRIPTION
**Pull request by ilooner**
This pull request is only intended for a review of dimensions computation it should not be merged.

### CODE REVIEW

File: library/pom.xml [hunk 1]
```diff
@@ -325,6 +325,11 @@
         </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>it.unimi.dsi</groupId>
+      <artifactId>fastutil</artifactId>
+      <version>6.6.3</version>
```
Comment (243826) [inline library/pom.xml]: 6.6.4?

File: library/pom.xml [hunk 2]
```diff
@@ -325,6 +325,11 @@
         </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>it.unimi.dsi</groupId>
+      <artifactId>fastutil</artifactId>
+      <version>7.0.6</version>
```
Comment (ilooner) [inline library/pom.xml]: Updated to latest fastutil version. When I started this project fastutil was on version 6.6.3 ...

File: library/src/main/java/com/datatorrent/lib/appdata/gpo/GPOUtils.java [hunk 3]
```diff
@@ -0,0 +1,2710 @@
...
+   * the fields.
+   * @param clazz The Class of the POJO to extract values from.
+   * @return An array of integer getters for given fields.
+   */
+  @SuppressWarnings({"unchecked","rawtypes"})
+  public static GetterInt<Object>[] createGetterInt(List<String> fields,
+                                                    Map<String, String> valueToExpression,
+                                                    Class<?> clazz)
+  {
+    GetterInt<Object>[] gettersInt = new GetterInt[fields.size()];
```
Comment (243826) [inline library/src/main/java/com/datatorrent/lib/appdata/gpo/GPOUtils.java]: here!

File: library/src/main/java/com/datatorrent/lib/appdata/gpo/GPOUtils.java [hunk 4]
```diff
@@ -0,0 +1,2710 @@
...
+        }
+      }
+    }
+
+    {
+      byte[] tempBytes = mutable.getFieldsByte();
+      GetterByte<Object>[] tempGetterByte = getters.gettersByte;
+
+      if(tempBytes != null) {
+        for(
```
Comment (243826) [inline library/src/main/java/com/datatorrent/lib/appdata/gpo/GPOUtils.java]: here2

Comment (243826) [inline library/src/main/java/com/datatorrent/lib/appdata/gpo/GPOUtils.java]: here3

Comment (243826) [inline library/src/main/java/com/datatorrent/lib/appdata/gpo/GPOUtils.java]: here4!

**Pull request closed without merging**